src/custom-element.js

```diff
@@ -659,7 +659,11 @@ function createBaseCustomElementClass(win) {
         const sizesAttr = this.getAttribute('sizes');
         this.sizeList_ = sizesAttr ? parseSizeList(sizesAttr) : null;
       }
-      if (this.sizeList_) {
+      let shouldSetWidth = !!this.sizeList_;
+      if (isExperimentOn(this.ampdoc_.win, 'amp-img-auto-sizes')) {
+        shouldSetWidth = !!this.sizeList_ && this.tagName !== 'AMP-IMG';
```

Comment:
The reason why I think we can do this is because the behavior (setting width via styles) that is removed is already implemented by properly propagating sizes on `amp-img`. So basically `sizes` on `<amp-img>` should after removing this, behave the same as all browsers default behavior with `sizes` on `<img>`. Reference: #17053, #8447

Reply:
Oh actually I think you are right. Because we have different layouts, the heuristic here would need to be a lot more nuanced.

Reply:
In that case, let me revert this change and rethink a solution to #17053, #8447.

Reply:
I think there right solution for #17053, #8447 is exactly auto sizes.

Essentially the story is:
_Assumption:_ `sizes` attribute in AMP and `img` have different semantic meaning. `sizes` in AMP is a way to use media queries to change layout. `sizes` on `img` is a way to pick a `srcset` that's not tied to viewport sizes.
1- Don't worry about `sizes` (the  `img` meaning`) we handle it for you
2- Use `sizes` on `amp-img` to defined `layout`. This has nothing to do with `sizes` on `img` (unfortunately naming, but too late to change. This should have been called `widths` to match `heights`)
3- remove this code but also do the `auto sizing` logic regardless of whether `sizes` was specified or not.(it should yield the same result anyway)

Reply:
This naming convention is SUPER confusing. =( 

> also do the `auto sizing` logic regardless of whether `sizes` was specified or not.

So basically, our answer to the developer is, you won't ever need to specify sizes on `AMP` because we do it for you automatically?

Reply:
Okay, summarizing our offline chat on this: 
1. We won't override existing cases of `sizes` on `<amp-img>`. These will set layout, and also have their `sizes` attribute propagated to the `<img>`. The expectation is that their sizes should end up being the same as the `sizes` attribute we generate anyway (*exception).
2. We will auto-generate sizes for all `<amp-img>` that don't have a `sizes` attribute specified. 

*exception is the case where due to bind / CSS toggles, the size of the `<amp-img>` changes without the the viewport changing in size, in this case, our autogenerated sizes will handle it correctly, and the developer-provided `sizes` attribute won't. 

Does that sound right to you @aghassemi ?